### PR TITLE
bug 1428330, collect metrics on buildbot perf via markus and datadog

### PR DESCRIPTION
**Pull request by Pike**
This adds timing metrics for:
- the actual automation task
- the time from change to end

This also adds gauge metrics keeping track of:
- how many builds are currently running (right now, 0 or 1)
- how many builds are pending

### CODE REVIEW

**Review by milescrabill — APPROVED** [review]
LGTM. Based on ericz's comments in the bug I think the defaults should be the same in your current infra as the future cloudops infra.

One note about timing() is that it looks like it expects milliseconds: https://github.com/mozilla-services/antenna/blob/master/antenna/breakpad_resource.py#L467-L470.

File: shared/shared.cfg [hunk 1]
```diff
@@ -1,12 +1,42 @@
...
+    {
+        'class': 'markus.backends.logging.LoggingMetrics',
+        'options': {
+            'logger_name': 'markus',
+            'leader': 'ELMO_METRICS',
+        }
+    }
+]
+if (
+        'ELMO_DATADOG_HOST' in os.environ and
```
[the remark this answers is not in the document]
Comment (Pike) [reply shared/shared.cfg]: I don't have datadog in my local setup in docker-compose (nor did I see how I would), so I'm using this on purpose to switch datadog on and off.

[the remark this answers is not in the document]
Comment (Pike) [reply shared/shared.cfg]: I've changed my mind on this one, and actually just use the datadog namespace as trigger, and leave the rest to defaults.

This is based on my change of mind what I'll do with the env config bug ;-)

**Comment by Pike** [comment]
This is squashed and merged, closing.

**Pull request closed without merging**